### PR TITLE
Limit number of logs recorded on the client side

### PR DESCRIPTION
**Pull request by zazabe**

### CODE REVIEW

**Comment by zazabe** [comment]
@njam please review

File: client-vendor/source/logger/handlers/recorder.js [hunk 1]
```diff
@@ -52,12 +53,17 @@ Recorder.prototype = {
...
+    var retention = this._options.retention;
+    var maxSize = this._options.recordMaxSize;
+    if (retention > 0) {
+      var retentionTime = this._getDate() - (retention * 1000);
       this._records = _.filter(this._records, function(record) {
         return record.date > retentionTime;
       });
     }
+    if (maxSize > 0 && this._records.length > maxSize) {
+      this._records = this._records.splice(this._records.length - maxSize, maxSize);
```
Comment (fvovan) [inline client-vendor/source/logger/handlers/recorder.js]: so we preserve older values.
won't `this._records = this._records.slice(-maxSize);` be simpler?

File: client-vendor/source/logger/handlers/recorder.js [hunk 2]
```diff
@@ -52,12 +53,17 @@ Recorder.prototype = {
    * @private
    */
   _cleanupRecords: function() {
-    if (this._options.retention) {
-      var retentionTime = this._getDate() - (this._options.retention * 1000);
+    var retention = this._options.retention;
+    var maxSize = this._options.recordMaxSize;
```
Comment (fvovan) [inline client-vendor/source/logger/handlers/recorder.js]: I would name it `recordMaxSize` to be consistent.

**Comment by zazabe** [comment]
@fvovan please check again

File: client-vendor/source/logger/handlers/recorder.js [hunk 3]
```diff
@@ -52,12 +53,17 @@ Recorder.prototype = {
...
+    var retention = this._options.retention;
+    var recordMaxSize = this._options.recordMaxSize;
+    if (retention > 0) {
+      var retentionTime = this._getDate() - (retention * 1000);
       this._records = _.filter(this._records, function(record) {
         return record.date > retentionTime;
       });
     }
+    if (recordMaxSize > 0 && this._records.length > recordMaxSize) {
+      this._records = this._records.splice(-recordMaxSize);
```
Comment (fvovan) [inline client-vendor/source/logger/handlers/recorder.js]: I meant `slice` not `splice`

Comment (zazabe) [inline client-vendor/source/logger/handlers/recorder.js]: ok, didn't read carefully ;)

**Pull request merged**